pkg/hooks/manager_test.go

```diff
@@ -93,35 +203,40 @@ var _ = Describe("HooksManager", func() {
 		It("Should find sidecar", func() {
 			hookPointName := hooksInfo.OnDefineDomainHookPointName
 
-			hookPath := filepath.Join(socketDir, "hook-sidecar-0")
-			os.MkdirAll(hookPath, os.ModePerm)
-			socketPath := filepath.Join(hookPath, "hook1.sock")
-			socket, err := hookListenAndServe(socketPath, "hook1", hookPointName, 0)
-			Expect(err).ToNot(HaveOccurred())
-			defer socket.Close()
-			defer os.Remove(socketPath)
+			t := newTestCase(socketDir, "hook1")
```

Comment:
btw: Use Gingko's temp dir feature for the socket dir?

Reply:
```diff
-           var err error
-           socketDir, err = os.MkdirTemp("", "hooksocketdir")
-           Expect(err).ToNot(HaveOccurred())
+           socketDir = filepath.Join(GinkgoT().TempDir(), "hooksocketdir")
```